same-tree/Leo.py

```diff
@@ -0,0 +1,14 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
+        if p and q:
+            return p.val == q.val and self.isSameTree(p.right, q.right) and self.isSameTree(p.left, q.left)
+        else:
+            return p == q
```

Comment:
nit: 가독성 측면에서 이게 쬐끔 더 친절한 코드가 되지 않을까 생각해보았습니다.

```suggestion
            return p == None and q == None
```

Reply:
전에 달레님이 말씀하신 간결함 보다는 이해가 잘가는 코드라고 말씀하신 예시일 수 있겠군요 ㅎㅎ

Reply:
네, 맞아요. 몇 자 더 타이핑해서 코드가 self-explanatory해질 수 있다면 인터뷰에서 이득이라고 생각합니다.

Reply:
이거 되게 신기하네요

`p == None and q == None` 을 입력하니 노란줄이 그어져서 `p is None and q is None` 을 하니 사라져요 ㅋㅋ 검색해보니 오른쪽이 좀 더 파이썬에서 권장되는 신택스라고 하네요. 컴파일에는 두 코드 다 문제없는거 같습니다 :)

Reply:
오 노란줄을 신경쓰면 자연스럽게 더 좋은
코드를 쓰게 될 수 있을 것 같네요

Reply:
제 실수네요 ㅋㅋ `is None`이 identity 검사이고, `== None`이 eqaulity 검사이니, 결과는 같아도 엄밀히 얘기해서 전자가 더 선호되겠네요. `None`은 유일무이하니까요. 이런 시비가 안 붙으려면 `not q and not q`로 처리할 수도 있겠네요. 😄